### PR TITLE
SQL: Improve performances of LTRIM/RTRIM

### PR DESCRIPTION
**Pull request by matriv**
Change custom stripping leading and trailing whitespaces implementation
to substantially improves performance:
```
Benchmark                         Mode  Cnt      Score     Error  Units
StringTrim.testWithStringBuilder  avgt   25  82547.575 ±  66.244  ns/op (existing impl)
StringTrim.testWithSubstring      avgt   25   1398.762 ± 101.152  ns/op (new impl)
StringTrim.testWithJavaStrip      avgt   25   1186.120 ±  10.374  ns/op (for reference)
```
Java's string stripLeading()/stripTrailing() not available to all supported JDKs.

Enhanced LENGTH unit tests and compine a couple of LTRIM/RTRIM integ
tests.

Relates to: #57594
(cherry picked from commit ee7868d68733f195dc46926a7eab3d9dd7033ef4)

### CODE REVIEW

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by costin** [comment]
While it's good to get performance in, I highly doubt in the overall scheme of things this makes any difference. Hence why I would minimize the overhead associated with such changes as the cost/benefit is low.

**Comment by matriv** [comment]
@costin The change makes the processing of a single string ~80x times faster, so I believe it's a substantial gain considering it could may have to be applied on thousands of rows.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/StringFunctionUtils.java [hunk 1]
```diff
@@ -47,11 +48,16 @@ static String trimTrailingWhitespaces(String s) {
...
+        for (int i = s.length() - 1; (i >= 0) && (endIdx < 0); i--) {
+            if (Character.isWhitespace(s.charAt(i)) == false) {
+                endIdx = i;
+            }
         }
-        return sb.toString();
+        if (endIdx < 0) {
+            return EMPTY;
+        }
+        return s.substring(0, endIdx + 1);
```
Comment (bpintea) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/StringFunctionUtils.java:60]: Nit: more compact alternative.

```
int i = s.length() - 1;
while (i >= 0 && Character.isWhitespace(s.charAt(i))) {
    i --;
}
return s.substring(0, i + 1);
```

Similar for treamLeading.

**Review by bpintea — APPROVED** [review]
LGTM, left minor alternative suggestion.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/StringFunctionUtils.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/StringFunctionUtils.java:60]: @bpintea cool, thx, with your approach:
```
Benchmark                         Mode  Cnt      Score    Error  Units
StringTrim.testWithJavaStrip      avgt   25   1342.251 ±  7.019  ns/op
StringTrim.testWithStringBuilder  avgt   25  82599.137 ± 86.073  ns/op
StringTrim.testWithSubstring      avgt   25   1035.607 ± 15.208  ns/op
```

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**